twitter/View Controllers/TimelineViewController.m

```diff
@@ -10,31 +10,51 @@
 #import "APIManager.h"
 #import "AppDelegate.h"
 #import "LoginViewController.h"
+#import "TweetCell.h"
+#import "Tweet.h"
+#import "UIImageView+AFNetworking.h"
 
-@interface TimelineViewController ()
+@interface TimelineViewController () <UITableViewDelegate,UITableViewDataSource>
+@property (weak, nonatomic) IBOutlet UITableView *twitterFeedTableView;
 - (IBAction)didTapLogout:(id)sender;
+@property (strong, nonatomic) NSMutableArray* arrayOfTweets;
```

Comment:
Does this need to be mutable? Again, a good rule of thumb is to make everything immutable (at least externally) unless it really needs to be mutable. If it needs to be mutable internally, you can redeclare the property in the .m file. For example:

```
// TimelineViewController.h
// ...
@property (strong, nonatomic, nonnull) NSArray<Tweet *> *tweets;
// ...
```
```
// TimelineViewController.m
@interface TimelineViewController ()
@property (strong, nonatomic, readonly, nonnull) NSMutableArray<Tweet *> *tweets;
@end
```